zerver/webhooks/gitlab/view.py

```diff
@@ -264,12 +264,13 @@ def get_object_url(payload: Dict[str, Any]) -> Text:
 @has_request_variables
 def api_gitlab_webhook(request: HttpRequest, user_profile: UserProfile,
                        stream: Text=REQ(default='gitlab'),
+                       topic: Text=REQ(default=None),
                        payload: Dict[str, Any]=REQ(argument_type='body'),
                        branches: Optional[Text]=REQ(default=None)) -> HttpResponse:
     event = get_event(request, payload, branches)
     if event is not None:
         body = get_body_based_on_event(event)(payload)
-        subject = get_subject_based_on_event(event, payload)
+        subject = topic or get_subject_based_on_event(event, payload)
```

Comment:
This line makes me chuckle, because I took the exact same approach when I first added a topic parameter to a webhook. This is the comment I received:

> I'd do this on its own line with if topic is None: (in general, we try to avoid use of falsyness in production code, because it combines together the totally different concepts of None and ''. Like, what if the user wanted to use a topic of ''? You'd need to think a while to figure out whether this code is broken in that case) and also add a test for it to maintain full test coverage.

If you're interested, this is the PR (in which I also used "expected_subject"...): https://github.com/zulip/zulip/pull/7471/files